locals.tf

```
@@ -21,14 +21,26 @@ locals {
   # for projects having explicit repositories definition
   repositories = flatten([
     for gcr in var.gcr_repositories : [
-      for repo in gcr.repositories : gcr.storage_region != null ? "${gcr.storage_region}.gcr.io/${gcr.project_id != null ? gcr.project_id : local.google_project_id}/${repo}" : "gcr.io/${gcr.project_id != null ? gcr.project_id : local.google_project_id}/${repo}"
+      for repo in gcr.repositories : gcr.storage_region != null
+        ? merge({
```

```suggestion
      ? merge(
        repo,
        {
          repo         = "${gcr.storage_region}.gcr.io/${gcr.project_id != null ? gcr.project_id : local.google_project_id}/${repo.repo}"
          grace        = repo.grace != null ? repo.grace : "0"
          allow_tagged = repo.allow_tagged != null ? repo.allow_tagged : false
          keep         = repo.keep != null ? repo.keep : 0
          tag_filter   = repo.tag_filter != null ? repo.tag_filter : ""
        }
      )
      : merge(
        repo,
        {
          repo         = "gcr.io/${gcr.project_id != null ? gcr.project_id : local.google_project_id}/${repo.repo}"
          grace        = repo.grace != null ? repo.grace : "0"
          allow_tagged = repo.allow_tagged != null ? repo.allow_tagged : false
          keep         = repo.keep != null ? repo.keep : 0
          tag_filter   = repo.tag_filter != null ? repo.tag_filter : ""
        }
      )
```
* need to switch maps order in order to keep elements from the 2end map
* need to handle null values